lg69t/firmware_tool.py

```diff
@@ -193,8 +195,7 @@ def Upgrade(port_name: str, bin_path: str, upgrade_type: UpgradeType, should_sen
         if not get_response(class_id, MSG_ID_FIRMWARE_ADDRESS, ser):
             return False
 
-        with open(bin_path, 'rb') as fd:
-            firmware_data = fd.read()
+        firmware_data = bin_file
```

Comment:
This is confusing, and might go back to something that could be clarified by the type hint. `firmware_data` used to be `bytes` but `bin_file` is a file handle right? Don't you still need:
```python
firmware_data = bin_file.read()
```